File: android/src/main/java/org/axen/flutter/glide/converter/AssetConverter.java
package org.axen.flutter.glide.converter;

import android.content.Context;
import android.content.res.AssetManager;
import android.graphics.BitmapFactory;

import androidx.annotation.NonNull;

import com.bumptech.glide.RequestBuilder;

import java.io.IOException;
import java.io.InputStream;

public final class AssetConverter<T> implements ImageConverter<T> {
    @NonNull
    public RequestBuilder<T> convert(
            @NonNull Context context,
            @NonNull RequestBuilder<T> builder,
            @NonNull Object resource
    ) throws IOException {
        AssetManager assetManager = context.getAssets();
        InputStream inputStream = assetManager.open((String)resource);
        return builder.load(BitmapFactory.decodeStream(inputStream));
    }
}

File: android/src/main/kotlin/org/axen/flutter/glide/converter/AssetConverter.kt
package org.axen.flutter.glide.converter

import android.content.Context
import android.content.res.AssetManager
import android.graphics.BitmapFactory
import com.bumptech.glide.RequestBuilder
import org.axen.flutter.glide.common.ImageConverter


class AssetConverter<T>: ImageConverter<T> {

    override fun convert(
        context: Context,
        builder: RequestBuilder<T>,
        resource: Any
    ): RequestBuilder<T> {
        val assetManager: AssetManager = context.assets
        val inputStream = assetManager.open(resource as String)
       return builder.load(BitmapFactory.decodeStream(inputStream))
    }
}